odoo/tools/misc.py

```diff
@@ -1600,3 +1600,17 @@ def hmac(env, scope, message, hash_function=hashlib.sha256):
         message.encode(),
         hash_function,
     ).hexdigest()
+
+
+class MockValidator:
```

Comment:
It's an object to be able to use a lambda to check a parameter value

Like
```python
self.assert_email_sent_smtp(
    # Must use the bounce address here because the mail server
    # is configured for the entire domain "test.com"
    smtp_from=MockValidator(lambda x: 'bounce' in x),
    message_from='sqli@test.com',
    from_filter='test.com',
    emails_count=8,
)
```

Moved to `odoo/odoo/tests/common.py`